### ISSUE TITLE
Autocomplete component in the styleguide dies on pick

### ISSUE THREAD

**Issue by clemensp** [pivotal-cf/pivotal-ui#472]
The error
---

https://gist.github.com/clemensp/f654386a840f6fec15b642c11cd7c9dc

How to reproduce
---

1. Go to: http://styleguide.cfapps.io/react_components_autocomplete.html
1. Open the web inspector in the browser
1. Scroll down to "Basic usage"
1. In the autocomplete box, type in "foo"
1. Click on the "foo" result

Result
---

[Stack trace](https://gist.github.com/clemensp/f654386a840f6fec15b642c11cd7c9dc
) in the console.

How this issue affects me
---

I get the same issue when trying out the autocompleter in my app.  :)

Please let me know if you need more information, or if you'd like me to test anything.
In the meantime, I'll spend a bit of time to see if it's something I can fix.

Cheers!

**Comment by clemensp** [comment]
Some extra info:

The version of pui-react-autocomplete that I have is v7.4.0.

It dies in node_modules/pui-react-autocomplete/autocomplete-list.js:51.

The code around that line is:

```
49: var onClick = function onClick(value, e) {
50:   e.preventDefault();
51:   undefined.props.onPick(value);
52: };
```

The error `Uncaught TypeError: Cannot read property 'props' of undefined` makes sense given the line `undefined.props.onPick(value);`

Looks like it goes wrong somewhere in the release process.

**Comment by clemensp** [comment]
Root cause: https://github.com/babel/babel/issues/730

Arrow functions inherit this from the outer scope.

The line that gets transpiled to undefined: https://github.com/pivotal-cf/pivotal-ui/blob/v7.4.0/library/src/pivotal-ui-react/autocomplete/autocomplete-list.js#L8

**Comment by clemensp** [comment]
A bunch of updates:

- This issue is only present in v7.4.0
  - In the development branch, it's been fixed.
- The commit that fixes it in development is [4f0110aa](https://github.com/pivotal-cf/pivotal-ui/commit/4f0110aaeadd56959dff5d4e228702139715f865)
- I tried putting together a pull request for 7.4.0, but it's just a tag, and not a branch, so I am not able to put one together.

Would you be able to back-port the fix and release v7.4.1?

**Comment by charleshansen** [comment]
Hey, thanks for the very detailed issue. We plan to publish 7.5.0 very soon, and this will have the fix. We don't have a good system for patching older minor releases (for a 7.4.1), but there shouldn't be any reason you can't upgrade to 7.5.0 when it comes out.

**Comment by clemensp** [comment]
Thanks for the quick response.  :)

I can definitely wait for the next release.

**Comment by jadekler** [comment]
Fixed in 7.5, published today!

**Comment by clemensp** [comment]
Thanks for the update.  :)
